pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -290,7 +292,18 @@ def _do_catch(self, pokemon, encounter_id, catch_rate_by_ball, is_vip=False):
                 current_ball += 1
             if items_stock[current_ball] == 0:
                 self.emit_event('no_pokeballs', formatted='No usable pokeballs found!')
-                break
+                if maximum_ball != ITEM_ULTRABALL and items_stock[ITEM_ULTRABALL] > min_ultraball_to_keep:
+                    maximum_ball = ITEM_ULTRABALL
+                    self.emit_event(
+                        'use_ultraball',
```

Comment:
This event is not necessary at all, we already have this:

```python
self.emit_event(
                'threw_pokeball',
                formatted='Used {ball_name}, with chance {success_percentage} ({count_left} left)',
                data={
                    'ball_name': self.item_list[str(current_ball)],
                    'success_percentage': self._pct(catch_rate_by_ball[current_ball]),
                    'count_left': items_stock[current_ball]
                }
            )
```

Which is enough information for the user.